File: C and C++/insertion_sort.cpp
#include<iostream>
using namespace std;

int *insertion_sort(int *arr,int n){
    
    for(int i=1;i<n;i++){

        int key=arr[i];
        int j=i-1;
        // increasing sorting
        // if we want decreasing order just change
        //arr[j]<key
        
        while(j>=0 && arr[j]>key){
            arr[j+1]=arr[j];
            j--;
        }
        arr[j+1]=key;

    }
    return arr;



}
int main(){


    
    int arr[]={2,0,9,3,60,4};
    int n=sizeof(arr)/sizeof(arr[0]);

    int *ptr=insertion_sort(arr,n);
    
    printf(" Insertion Sort\n");

    for(int i=0;i<n;i++){
        cout<<*(ptr+i)<<" ";

    }


}

File: Python/Sorting/insertion_sort.py
def insertion_sort(arr):
    for i in range(1,len(arr)):
        key=arr[i]
        j=i-1
        while j>=0 and arr[j] > key:
            arr[j+1]=arr[j]
            j=j-1
        arr[j+1]=key


arr=[5,4,2,3,66,44,77]
print("Before sorting")
for i in range(0,len(arr)):
    print(arr[i],end=' ')
print("\n")    

print("After Sorting")
insertion_sort(arr)
for i in range(0,len(arr)):
    print(arr[i],end=' ')
